tests/helpers/test-methods.js

```diff
@@ -138,4 +138,32 @@ module.exports = {
       editorInstance.setData(textAreaContent);
     }, { locator, textAreaContent });
   },
+  /**
+  * Waits for Nuxt 3 page to be fully hydrated.
+  * This function is polling for status of nuxtApp.isHydrating.
+  * @param {Page} page The page to wait for.
+  */
+  waitForNuxtHydration: async (page) => {
+    const timeout = 3000;
+    const pollInterval = 100;
+    const startTime = Date.now();
+
+    while (Date.now() - startTime < timeout) {
+      const isHydrated = await page.evaluate(() => {
+        const nuxtApp = window.useNuxtApp && window.useNuxtApp();
+        if (nuxtApp && typeof nuxtApp.isHydrating === 'boolean') {
+          return nuxtApp.isHydrating === false;
+        }
+        return false;
+      });
+
+      if (isHydrated) {
+        return;
+      }
+  
+      await new Promise(resolve => setTimeout(resolve, pollInterval));
+    }
+
+    throw new Error(`Nuxt hydration failed to complete within ${timeout}ms`);
+  },
```

Comment:
similar code to https://github.com/nuxt/test-utils/blob/49ccb2f955215659c10f63fb93ee2c0fcf67581e/src/e2e/browser.ts#L69C9-L69C9

couldn't use waitForFunction as that fails CSP for unsafe-eval, implemented a similar function with polling

Reply:
unfortunately, this is problematic since the the timeout makes it unreliable. we should not have a timeout beside the timeout that is configured within playwright. 

Can we avoid unsafe-eval by adding the 1-line helpers as public API from our JS? Can we still simply call stuff from playwright? e.g. document.lupuswaitforNuxtHydrate() and make it a promise?
Or can we expose an event from Nuxt - as shown in my nuxtjs-drupal-ce code- and directly listen to do that with playwright code? We need to solve this to properly interact with the JS code of the app once so we have a proper solution that we can repeat then.

So let's find out what we can still do with unsafe-eval.